pkg/cloudprovider/vsphere/nodemanager.go

```diff
@@ -93,48 +96,202 @@ func (nm *NodeManager) removeNode(node *v1.Node) {
+				if err != nil {
+					glog.Error("Discovering node error dc:", err)
+					setGlobalErr(err)
+					continue
+				}
+			} else {
+				datacenters := strings.Split(vsi.Cfg.Datacenters, ",")
+				for _, dc := range datacenters {
+					dc = strings.TrimSpace(dc)
+					if dc == "" {
+						continue
+					}
+					datacenterObj, err := vclib.GetDatacenter(ctx, vsi.Conn, dc)
+					if err != nil {
+						glog.Error("Discovering node error dc:", err)
+						setGlobalErr(err)
+						continue
+					}
+					datacenterObjs = append(datacenterObjs, datacenterObj)
+				}
+			}
+
+			for _, datacenterObj := range datacenterObjs {
+				found := getVMFound()
+				if found == true {
+					break
+				}
+
+				glog.V(4).Infof("Finding node %s in vc=%s and datacenter=%s", myNodeID, vc, datacenterObj.Name())
+				queueChannel <- &vmSearch{
+					vc:         vc,
+					datacenter: datacenterObj,
+				}
+			}
+		}
+		close(queueChannel)
+	}()
+
+	for i := 0; i < cm.POOL_SIZE; i++ {
+		go func() {
+			for res := range queueChannel {
+				var vm *vclib.VirtualMachine
+				var err error
+				if searchBy == FindVMByUUID {
+					vm, err = res.datacenter.GetVMByUUID(ctx, myNodeID)
+				} else {
+					vm, err = res.datacenter.GetVMByDNSName(ctx, myNodeID)
+				}
+
+				if err != nil {
+					glog.Errorf("Error while looking for vm=%+v in vc=%s and datacenter=%s: %v",
+						vm, res.vc, res.datacenter.Name(), err)
+					if err != vclib.ErrNoVMFound {
+						setGlobalErr(err)
+					} else {
+						glog.V(2).Infof("Did not find node %s in vc=%s and datacenter=%s",
+							myNodeID, res.vc, res.datacenter.Name())
+					}
+					continue
+				}
+
+				var oVM mo.VirtualMachine
+				err = vm.Properties(ctx, vm.Reference(), []string{"config", "summary", "summary.config", "guest.net", "guest"}, &oVM)
+				if err != nil {
+					glog.Errorf("Error collecting properties for vm=%+v in vc=%s and datacenter=%s: %v",
+						vm, res.vc, res.datacenter.Name(), err)
+					continue
+				}
+
+				addrs := []v1.NodeAddress{}
+				for _, v := range oVM.Guest.Net {
+					for _, ip := range v.IpAddress {
+						if net.ParseIP(ip).To4() != nil {
+							v1helper.AddToNodeAddresses(&addrs,
+								v1.NodeAddress{
+									Type:    v1.NodeExternalIP,
+									Address: ip,
+								}, v1.NodeAddress{
+									Type:    v1.NodeInternalIP,
+									Address: ip,
+								}, v1.NodeAddress{
+									Type:    v1.NodeHostName,
+									Address: oVM.Guest.HostName,
+								},
+							)
+						}
+					}
+				}
+
+				glog.V(2).Infof("Found node %s as vm=%+v in vc=%s and datacenter=%s",
+					nodeID, vm, res.vc, res.datacenter.Name())
+				glog.V(2).Info("Hostname: ", oVM.Guest.HostName, " UUID: ", oVM.Summary.Config.Uuid)
+
+				nodeInfo := &NodeInfo{dataCenter: res.datacenter, vm: vm, vcServer: res.vc,
+					UUID: oVM.Summary.Config.Uuid, NodeName: oVM.Guest.HostName, NodeAddresses: addrs}
+				nm.addNodeInfo(nodeInfo)
+				for range queueChannel {
```

Comment:
Yea, I noticed it a long long time ago and it's one of those things that scares me to remove... like in case it is a timing thing. hahah! will remove and cross my fingers! LOL